components/site/src/lib.rs

```diff
@@ -864,9 +864,44 @@ impl Site {
         context.insert("taxonomies", &taxonomies);
         context.insert("config", &self.config);
 
-        let sitemap = &render_template("sitemap.xml", &self.tera, context, &self.config.theme)?;
+        // Count total number of urls to include in sitemap
+        let total_number = pages.len() + sections.len() + taxonomies.len();
+        if total_number > self.config.sitemap_limit {
+            // Split the sitemap and reference all sitemaps in a main sitemap
+
+            // Group all sitemap entries in one vector
+            let mut all_sitemap_entries = Vec::new();
+            all_sitemap_entries.append(&mut pages);
+            all_sitemap_entries.append(&mut sections);
+            for terms in taxonomies {
+                let mut terms = terms;
+                all_sitemap_entries.append(&mut terms);
+            }
+            
+            // Slice the vector in chunks and create sitemap file for each
+            let mut xml_files = Vec::new();
+            for (i, chunk) in all_sitemap_entries.chunks(self.config.sitemap_limit).enumerate() {
+                let mut chunk_context = context.clone();
```

Comment:
Instead of doing that, can you delay inserting into the context?
In short, do an early return of the fn if the number of entries is lower than the `sitemap_limit` where you can insert into the context.

Then you can do a big vec of sitemap entries and chunk that, only creating a context when you need to render one.

I'm also thinking whether we can do a breaking change and have only one entry in context `entries` that would simplify things. since there's no need to differentiate between pages/sections/taxonomies there afaik. But that's something I would do afterwards, don't worry about it.

Reply:
I gave it a try, did I understand you right?
thanks for your comments.